test/Microsoft.AspNetCore.Http.Connections.Tests/WebSocketsTests.cs

```diff
@@ -24,10 +24,18 @@ public WebSocketsTests(ITestOutputHelper output)
         {
         }
 
+        // For unknown reasons using the built-in WebSocketMessageType as a parameter
+        // causes "Non-serializable data" warnings in xunit
+        public enum TestWebSocketMessageType
+        {
+            Binary = 1,
+            Text = 0
+        }
+
         [Theory]
-        [InlineData(WebSocketMessageType.Text)]
-        [InlineData(WebSocketMessageType.Binary)]
-        public async Task ReceivedFramesAreWrittenToChannel(WebSocketMessageType webSocketMessageType)
+        [InlineData(TestWebSocketMessageType.Text)]
+        [InlineData(TestWebSocketMessageType.Binary)]
```

Comment:
As I suggested on Twitter, I believe we convert string data into enum values. I don't know if `nameof` supports enum values (to help preserve values across renames), but given your work-around, you've already given up on rename refactor support anyway...

Reply:
`nameof` should work here actually, so we could try that.